src/praisonai-agents/praisonaiagents/tools/registry.py

```
@@ -409,54 +420,80 @@ def get_all(self) -> Dict[str, Union[BaseTool, Callable]]:
     
+    def _entry_points_for_group(self, group: str) -> list:
+        """Fetch entry points for a group, tolerating the Python 3.9 API shape."""
+        try:
+            # Python 3.10+ style
+            return list(_get_entry_points()(group=group))
+        except TypeError:
+            # Python 3.9 fallback
+            try:
+                all_eps = _get_entry_points()()
+                return list(all_eps.get(group, []))
+            except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)